src/libraries/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -49,6 +49,20 @@ private static void ProcessStartupHooks(string diagnosticStartupHooks)
                 startupHookParts.AddRange(startupHooksVariable.Split(Path.PathSeparator));
             }
 
+            // Process each startup hook
+            for (int i = 0; i < startupHookParts.Count; i++)
+            {
+                CallStartupHook(startupHookParts[i]);
+            }
+        }
+
+        private static unsafe void CallStartupHook(char* pStartupHookPart)
+        {
+            CallStartupHook(new string(pStartupHookPart));
```

Comment:
Was this small helper to encapsulate the unsafe?

Reply:
Yes, it allows passing the assembly path buffer directly instead of attempting to convert it to a managed string from native code.

Reply:
Actually - I see this is needed for the call.